sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -132,6 +132,12 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
     }
 #endif
     if (timeout != 0) {
+#if defined(DEVELHELP) && IS_ACTIVE(SOCK_HAS_ASYNC)
+        if (reg->async_cb.generic) {
+            LOG_WARNING("gnrc_sock: timeout != 0 with async callback set may lead "
+                        "to unexpected behavior.\n");
```

Comment:
I wouldn't know what unexpected behavior to expect here

Reply:
Made the wording a bit more specific.